crates/uv-cli/src/lib.rs

```diff
@@ -2932,6 +2932,16 @@ pub struct RunArgs {
     /// By default, environment modifications are omitted, but enabled under `--verbose`.
     #[arg(long, env = EnvVars::UV_SHOW_RESOLUTION, value_parser = clap::builder::BoolishValueParser::new(), hide = true)]
     pub show_resolution: bool,
+
+    /// Number of times this process has been recursively invoked by uv
+    /// run. Used to detect and error on potential infinite recursion.
+    #[arg(long, hide = true, env = EnvVars::UV_RUN_RECURSION_DEPTH)]
```

Comment:
I marked both of these as hidden b/c they didn't seem useful enough to users to warrant polluting the `uv run` output. I think the `RECURSION_DEPTH` argument at least should probably stay hidden. I could see an argument for `max_recursion_depth` being made visible in case someone is implementing some weird recursive algorithm with uv subprocesses and legitimately needs to raise the limit, but that seems pretty unlikely to me as well.

Reply:
I think it'd make sense to read this directly without using Clap / adding a flag to the CLI.